lib/Onyx.js

```diff
@@ -1082,15 +1082,18 @@ function multiSet(data) {
 function applyMerge(existingValue, changes) {
     const lastChange = _.last(changes);
 
-    if (_.isArray(existingValue) || _.isArray(lastChange)) {
+    if (_.isArray(lastChange)) {
         return lastChange;
     }
 
-    if (_.isObject(existingValue) || _.every(changes, _.isObject)) {
-        // Object values are merged one after the other
-        // lodash adds a small overhead so we don't use it here
-        // eslint-disable-next-line prefer-object-spread, rulesdir/prefer-underscore-method
-        return _.reduce(changes, (modifiedData, change) => utils.fastMerge(modifiedData, change),
+    if (_.some(changes, _.isObject)) {
+        // There might be nullish values in a batch of changes that we want to ignore when merging objects,
+        // because otherwise they would overwrite the previous changes.
```

Comment:
```suggestion
        // because otherwise they overwrite the previous changes in the batch.
```
Just adding a little clarity.